### PR TITLE
added recipie for amgcl

### PR DESCRIPTION
**Pull request by davidsanfal**
Specify library name and version:  **amgcl/1.4.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/amgcl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class UncrustifyConan(ConanFile):
+    name = "amgcl"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ddemidov/amgcl"
+    topics = ("conan", "amgcl", "mathematics")
+    license = "MIT"
+    description = "AMGCL is a header-only C++ library for solving large sparse linear systems with algebraic multigrid (AMG) method."
+    settings = "compiler"
+    no_copy_source = True
+    requires = [("boost/1.76.0")]
```
Comment (SSE4) [inline recipes/amgcl/all/conanfile.py:14]: sounds like an excessive syntax, but okay...

File: recipes/amgcl/all/test_package/solver.cpp [hunk 2]
```diff
@@ -0,0 +1,648 @@
+#include <iostream>
```
Comment (SSE4) [inline recipes/amgcl/all/test_package/solver.cpp:1]: example is pretty large... did you take it somewhere?

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — COMMENTED** [review]
Nice project to be included, it can be expanded in the future.

Consider avoiding "master" branch for PRs. If you need to open a new PR, it would create a conflict in your repository.

Please, consider https://github.com/davidsanfal/conan-center-index/pull/6 which reduced your test_package. Also, added optional packages (not all available on CCI)

**Comment by prince-chrismc** [comment]
There's also CMake targets that are installed! https://github.com/ddemidov/amgcl/blob/681bfc71edf6c3721109bd0f263b33c92ec3ba2e/CMakeLists.txt#L303

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
> There's also CMake targets that are installed! https://github.com/ddemidov/amgcl/blob/681bfc71edf6c3721109bd0f263b33c92ec3ba2e/CMakeLists.txt#L303

can be added later in a follow-up

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]
Let's merge this one! 🚀

**Pull request merged**